pkg/common/telemetry/latency.go

```diff
@@ -0,0 +1,80 @@
+package telemetry
+
+import (
+	"sync"
+	"time"
+)
+
+// Latency is used to track timing between two specific events. It
+// is a generic version of CallCounter and can be used to measure latency between any two events.
+//
+// Example 1:
+//		func Foo() {
+//		    latency := StartLatencyMetric(metrics, "foo")
+//		 	call.AddLabel("food", "burgers")
+//  		// do something
+//	     	latency.Measure() // measure time elapsed between StartLatencyMetric() and Measure()
+//			// do other things
+//		}
+//
+// Example 2:
+//		func Bar() {
+//		    latency := StartLatencyMetric(metrics, "bar")
+//		 	call.AddLabel("food", "pizza")
+//  		// do something
+//	     	latency.MeasureAndReset() // emits metric for time elapsed between StartLatencyMetric() and Measure()
```

Comment:
Is there a use case you had in mind for this method?

Reply:
This was mainly for making the `*Latency` object re-usable at multiple points within a function or an API call. I'm not sure if it's needed immediately for any use-case so I'm remove it for now and think about it as a future enhancement if needed.